cpp/src/arrow/compute/kernels/scalar_arithmetic.cc

```diff
@@ -1606,6 +1774,20 @@ const FunctionDoc trunc_doc{
     ("Calculate the nearest integer not greater in magnitude than to the "
      "argument element-wise."),
     {"x"}};
+
+const FunctionDoc round_doc{
+    "Round the arguments element-wise",
+    ("Options are used to control the rounding mode and number of digits.\n"
+     "Default behavior is to round to nearest integer."),
+    {"x"},
+    "RoundOptions"};
+
+const FunctionDoc mround_doc{
```

Comment:
nit: perhaps RoundToMultiple (and possibly RoundToPlace) might be clearer than round/mround (unless mround is a common name? I see Excel uses it at least)

Reply:
I am still a bit confused on how to name functions as I have seen reviews on both camps:
* use the short name commonly used by other tools
* use a long form that is explicit and unambiguous

Do you refer solely to the C++ class/variables or also to the name used with `CallFunction`?
`mround` does comes from Excel/Office-like tools.

I am ok with being explicit in the C++ code using `RoundToMultiple` and `RoundToPlace`. Do you also suggest having the name for `CallFunction` be `round_to_multiple` and `round_to_place`?

Reply:
I was talking about the name used with CallFunction, sorry.

I am ok with round/mround, you are right in that there's not too much consistency. Just a thought since it took a moment for mround to click, but of course I'm not so familiar with Excel et al.